jwst/outlier_detection/outlier_detection_ifu.py

```diff
@@ -0,0 +1,174 @@
+    the various functions and then controls the operation of this process
+    through all the steps used for the detection.
+
+    Notes
+    -----
+    This routine performs the following operations::
+
+      1. Extracts parameter settings from input model and merges
+         them with any user-provided values
+      2. Resamples all input images into grouped observation mosaics.
+      3. Creates a median image from all grouped observation mosaics.
+      4. Blot median image to match each original input image.
+      5. Perform statistical comparison between blotted image and original image
+         to identify outliers.
+      6. Updates input data model DQ arrays with mask of detected outliers.
+
+    """
+    DEFAULT_SUFFIX = 's3d'
+
+    def __init__(self, input_models, 
+                    reffiles=None,  
+                    **pars):
+        """
+        Parameters
+        ----------
+        input_models : list of DataModels, str
+            list of data models as ModelContainer or ASN file,
+            one data model for each input image
+
+        drizzled_models : list of objects
+            ModelContainer containing drizzled grouped input images
+
+        reffiles : dict of `jwst.datamodels.DataModel`
+            Dictionary of datamodels.  Keys are reffile_types.
+                        
+            
+        """
+        OutlierDetection.__init__(self, input_models, reffiles=reffiles, **pars)
+        # NOTE:  Need to confirm that this attribute accurately reports the
+        #        channel 'names' for both types of IFU data; MIRI and NRS
+        self.channels = self.input_models[0].meta.instrument.channel
+
+    def do_detection(self):
+        """Flag outlier pixels in DQ of input images
+        """
+        self.build_suffix(**self.outlierpars)
+        save_intermediate_results = self.outlierpars['save_intermediate_results']
+
+        # start by creating copies of the input data to place the separate
+        # data in after blotting the median-combined cubes for each channel
+        self.blot_models = self.input_models.copy()
+        for model in self.blot_models:
+            # replace arrays with all zeros to accommodate blotted data
+            model.data = np.zeros(model.data.shape,dtype=model.data.dtype) 
+            
+        # Create the resampled/mosaic images for each group of exposures
+        #
+        log.info("Performing IFU outlier_detection for exptype {}".format(self.exptype))
+        for channel in range(len(self.channels)):
+            ch = self.channels[channel]
+            
+            cubestep = CubeBuildStep(config_file=cube_build_config,
+                                        channel=ch,
+                                        single='true')
+            single_IFUCube_result = cubestep.process(self.input_models)
+
+            for model in single_IFUCube_result:
+                model.meta.filename += self.resample_suffix
```

Comment:
Same comment as before regarding appending a suffix to an existing filename.

Reply:
Again, if this assumption is not valid, a more general operation can be defined.